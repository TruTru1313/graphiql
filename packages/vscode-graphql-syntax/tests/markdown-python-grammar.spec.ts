import { tokenizeFile } from './__utilities__/utilities';

describe('inline.graphql.markdown.codeblock grammar python', () => {
  const scope = 'inline.graphql.markdown.codeblock';

  it('should tokenize a markdown file with python', async () => {
    const result = await tokenizeFile('__fixtures__/test-py.md', scope);
    expect(result).toMatchSnapshot();
  });
});
