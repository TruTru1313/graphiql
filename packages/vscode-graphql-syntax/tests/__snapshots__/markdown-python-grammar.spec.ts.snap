// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inline.graphql.markdown.codeblock grammar python > should tokenize a markdown file with python 1`] = `
# Python      | 
              | 
\`\`\`           | markup.fenced_code.block.python punctuation.definition.markdown
Python        | markup.fenced_code.block.python fenced_code.block.language.markdown
              | markup.fenced_code.block.python meta.embedded.block.python
query =       | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
gql           | markup.fenced_code.block.python meta.embedded.block.python entity.name.function
(             | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
"""           | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
query         | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
getContinents | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
"""           | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql string.quoted.multi.python
)             | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
query =       | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
gql           | markup.fenced_code.block.python meta.embedded.block.python entity.name.function
(             | markup.fenced_code.block.python meta.embedded.block.python
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
query         | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
getContinents | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
)             | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
query =       | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
gql           | markup.fenced_code.block.python meta.embedded.block.python entity.name.function
(             | markup.fenced_code.block.python meta.embedded.block.python
'''           | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
query         | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql
getContinents | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
'''           | markup.fenced_code.block.python meta.embedded.block.python meta.embedded.block.graphql string.quoted.multi.python
)             | markup.fenced_code.block.python meta.embedded.block.python
              | markup.fenced_code.block.python meta.embedded.block.python
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
#graphql      | markup.fenced_code.block.python meta.embedded.block.python comment.line.graphql.js
              | markup.fenced_code.block.python meta.embedded.block.python
query         | markup.fenced_code.block.python meta.embedded.block.python keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python
getContinents | markup.fenced_code.block.python meta.embedded.block.python entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python
query =       | markup.fenced_code.block.python meta.embedded.block.python
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
#graphql      | markup.fenced_code.block.python meta.embedded.block.python comment.line.graphql.js
              | markup.fenced_code.block.python meta.embedded.block.python
query         | markup.fenced_code.block.python meta.embedded.block.python keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python
getContinents | markup.fenced_code.block.python meta.embedded.block.python entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
'''           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python
"""           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
#graphql      | markup.fenced_code.block.python meta.embedded.block.python comment.line.graphql.js
              | markup.fenced_code.block.python meta.embedded.block.python
query         | markup.fenced_code.block.python meta.embedded.block.python keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python
getContinents | markup.fenced_code.block.python meta.embedded.block.python entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
"""           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
              | markup.fenced_code.block.python meta.embedded.block.python
"""           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
#graphql      | markup.fenced_code.block.python meta.embedded.block.python comment.line.graphql.js
              | markup.fenced_code.block.python meta.embedded.block.python
query         | markup.fenced_code.block.python meta.embedded.block.python keyword.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python
getContinents | markup.fenced_code.block.python meta.embedded.block.python entity.name.function.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql
continents    | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
{             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
code          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
name          | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql variable.graphql
              | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql meta.selectionset.graphql punctuation.operation.graphql
}             | markup.fenced_code.block.python meta.embedded.block.python meta.selectionset.graphql punctuation.operation.graphql
"""           | markup.fenced_code.block.python meta.embedded.block.python string.quoted.multi.python
\`\`\`           | markup.fenced_code.block.python punctuation.definition.markdown
              | 
`;
